plugin-java/che-plugin-java-ext-jdt/che-jdt-ext-machine/src/main/java/org/eclipse/che/jdt/rest/RefactoringService.java

```diff
@@ -137,6 +139,14 @@ public void changeChangeEnabledState(ChangeEnabledState state) throws Refactorin
         manager.changeChangeEnabled(state);
     }
 
+    @POST
+    @Path("change/preview")
+    @Produces("application/json")
+    @Consumes("application/json")
+    public ChangePreview getChangePreview(RefactoringChange change) throws RefactoringException, CoreException {
```

Comment:
How are you deal with RefactoringException, CoreException? What response will be?